src/store/index.ts

```diff
@@ -1,34 +1,33 @@
-import { Folder } from '../types/Folder'
-import { Feed } from '../types/Feed'
-import feeds, { FEED_ACTION_TYPES } from './feed'
-import folders, { FOLDER_ACTION_TYPES } from './folder'
-import { FOLDER_MUTATION_TYPES, FEED_MUTATION_TYPES, FEED_ITEM_MUTATION_TYPES } from '../types/MutationTypes'
+import { FOLDER_MUTATION_TYPES, FEED_MUTATION_TYPES, FEED_ITEM_MUTATION_TYPES, APPLICATION_MUTATION_TYPES } from '../types/MutationTypes'
+import feeds, { FEED_ACTION_TYPES, FeedState } from './feed'
+import folders, { FOLDER_ACTION_TYPES, FolderState } from './folder'
 import items, { FEED_ITEM_ACTION_TYPES, ItemState } from './item'
+import app, { APPLICATION_ACTION_TYPES, AppInfoState } from './app'
 
 export const MUTATIONS = {
+	...APPLICATION_MUTATION_TYPES,
 	...FEED_MUTATION_TYPES,
 	...FOLDER_MUTATION_TYPES,
 	...FEED_ITEM_MUTATION_TYPES,
 }
 
 export const ACTIONS = {
+	...APPLICATION_ACTION_TYPES,
 	...FEED_ACTION_TYPES,
 	...FOLDER_ACTION_TYPES,
 	...FEED_ITEM_ACTION_TYPES,
 }
 
-export type AppState = {
-  feeds: Feed[];
-  folders: Folder[];
-} & ItemState;
+export type AppState = FolderState & FeedState & ItemState & AppInfoState;
 
 type Func = (name: string, value: unknown) => void;
-export type ActionParams = { commit: Func; dispatch: Func; state: AppState };
+export type ActionParams<T> = { commit: Func; dispatch: Func; state: T };
```

Comment:
realized we don't have access to the state from different modules inside a vuex module (e.g. can't touch feeds from inside the folders module), so I made this type generic to make that more obvious during development